diff --git a/node_modules/browserslist/index.js b/node_modules/browserslist/index.js
index 401878d..0a875e1 100644
--- a/node_modules/browserslist/index.js
+++ b/node_modules/browserslist/index.js
@@ -368,7 +368,6 @@ function browserslist (queries, opts) {
     mobileToDesktop: opts.mobileToDesktop
   }
 
-  env.oldDataWarning(browserslist.data)
   var stats = env.getStat(opts, browserslist.data)
   if (stats) {
     context.customUsage = { }
